srml/staking/src/lib.rs

```diff
@@ -514,50 +711,19 @@ impl<T: Trait> Module<T> {
 			});
 		}
 
-		let event = if new_slash_count > grace {
-			let slash = {
-				let base_slash = Self::current_offline_slash();
-				let instances = slash_count - grace;
-
-				let mut total_slash = BalanceOf::<T>::default();
-				for i in instances..(instances + count as u32) {
-					if let Some(total) = base_slash.checked_shl(i)
-							.and_then(|slash| total_slash.checked_add(&slash)) {
-						total_slash = total;
-					} else {
-						// reset slash count only up to the current
-						// instance. the total slash overflows the unit for
-						// balance in the system therefore we can slash all
-						// the slashable balance for the account
-						<SlashCount<T>>::insert(v.clone(), slash_count + i);
-						total_slash = Self::slashable_balance(&v);
-						break;
-					}
-				}
-
-				total_slash
-			};
+		let prefs = Self::validators(&v);
+		let unstake_threshold = prefs.unstake_threshold.min(MAX_UNSTAKE_THRESHOLD);
+		let max_slashes = grace + unstake_threshold;
 
+		let event = if new_slash_count > max_slashes {
```

Comment:
Should this be `>=`?

i.e. `unstake_threshold = 3`, `grace = 0`, should they be slashed at `3` or `4` offline reports?

Reply:
In my mind actually at `4` slash should happen with your example, hence still `>`. 

With that perspective wouldn't someone with `grace=0` and`unstake_threshold=0` be slashed by default?

Reply:
This makes sense, look at other comments below